routes/api/contacts.js

```diff
@@ -1,25 +1,137 @@
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+// GET contacts list =================================
+router.get("/contacts", async (_, res) => {
+  try {
+    const contacts = await listContacts();
+    if (contacts.length > 0) {
+      res
+        .status(200)
+        .json({ status: "200", message: "Success", data: { contacts } })
+    } else {
+      res
+        .status(404)
+        .json({ status: "404", message: "Contacts not found" });
+    }
+  } catch (error) {
+    res
+      .status(404)
+      .json({ status: "404", message: "Not found", details: error.message });
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+// GET contact by id =================================
+router.get("/contacts/:contactId", async (req, res) => {
+  try {
+    const { contactId } = req.params;
+    const contact = await getContactById(contactId);
+    if (contact) {
+      res
+        .status(200)
+        .json({ status: "200", message: "Success", data: { contact } });
+    } else {
+      res
+        .status(404)
+        .json({ status: "404", message: "Contact not found" });
+    }
+  } catch (error) {
+    res
+      .status(404)
+      .json({ status: "404", message: "Not found", details: error.message});
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+// DELETE contact by id =================================
+router.delete("/contacts/:contactId", async (req, res) => {
+  try {
+    const { contactId } = req.params;
+    const contact = await getContactById(contactId);
+    if (contact) {
+      await removeContact(contactId);
+      res
+        .status(201)
+        .json({ status: "201", message: "Contact deleted" });
+    } else {
+      res
+        .status(404)
+        .json({ status: "404", message: "Contact not found" });
+    }
+  } catch (error) {
+    res
+      .status(400)
+      .json({ status: "404", message: "Not found", details: error.message });
+  }
+});
 
-module.exports = router
+// ADD contact =================================
+router.post("/contacts", async (req, res) => {
+  try {
+    const result = validateContact(req.body);
+    if (result.error) {
+      res
+        .status(400)
+        .json({ status: "400", message: result.error.message });
+    } else {
+      const validatedContact = result.value;
+      validatedContact.id = uuidv4();
+      await addContact(validatedContact);
+      res
+        .status(201)
+        .json({ status: "201", message: "Contact added", data: validatedContact });
+    }
+  } catch (error) {
+    res
+      .status(400)
+      .json({ status: "404", message: "Not found", details: error.message });
+  }
+});
+
+// UPDATE contact =================================
+router.put("/contacts/:contactId", async (req, res) => {
+  try {
+    if (Object.keys(req.body) === 0) {
```

Comment:
Nie musisz tego robić, bo Joi Ci to sprawdzi.

Reply:
W joi użyłaś required, wiec pola są wymagane.

Reply:
Wiem, że Joi to by wyłapał, ale w zadaniu było:
"jeżeli nie ma body, zwraca json z kluczem {"message": "missing fields"} i statusem 400".
Więc dodałam to sprawdzanie, żeby na pewno to był wskazany komunikat.
Poprawione.